Error when building the code

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

## Description
When I run the build process exactly as listed in the Quickstart, I get the following results:
```
/path/to/navigation/LieGroupEKF.h:98:46: error: incomplete type ‘gtsam::traits<Eigen::Matrix<double, -1, -1> >’ used in nested name specifier
   98 |         const Matrix nextX = traits<Matrix>::Retract(this->X_, xi * dt, A ? &Dexp : nullptr); // just addition
      |                                              ^~~~~~~
make[2]: *** [examples/CMakeFiles/IEKF_NavstateExample.dir/build.make:76: examples/CMakeFiles/IEKF_NavstateExample.dir/IEKF_NavstateExample.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:24572: examples/CMakeFiles/IEKF_NavstateExample.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

```
<!-- A clear description of the bug -->

## Steps to reproduce

1. Run the quickstart installation instructions on the latest branch of develop

<!-- If you have a code sample, unit test, error messages, stack traces, etc., please provide it here as well -->

## Expected behavior

The build should succeed.

## Environment
1. gcc version 11.4
2. boost 1.7.4
3. cmake 3.22.1

<!-- Please provide information about your code environment, things such as OS, language of use (C++, Matlab or Python), version of dependent libraries if using a custom build etc., anything to provide us more information. -->

<!-- The more the information, the faster we can help resolve the issue -->

## Additional information
Not using system Eigen, using the GTSAM bundled one
<!-- Add any other infor or context about the problem here. -->

--- a/examples/GEKF_Rot3Example.cpp
+++ b/examples/GEKF_Rot3Example.cpp
@@ -10,7 +10,7 @@
  * -------------------------------------------------------------------------- */
 
 /**
- * @file IEKF_Rot3Example.cpp
+ * @file GEKF_Rot3Example.cpp
  * @brief Left‐Invariant EKF on SO(3) with state‐dependent pitch/roll control
  * and a single magnetometer update.
  * @date April 25, 2025

--- a/examples/IEKF_NavstateExample.cpp
+++ b/examples/IEKF_NavstateExample.cpp
@@ -17,6 +17,7 @@
  */
 
 #include <gtsam/base/Matrix.h>
+#include <gtsam/base/VectorSpace.h>
 #include <gtsam/base/OptionalJacobian.h>
 #include <gtsam/navigation/InvariantEKF.h>
 #include <gtsam/navigation/NavState.h>
@@ -73,23 +74,23 @@ int main() {
   z2 << 0.6, 0, 0;
 
   cout << "=== Init ===\nX: " << ekf.state() << "\nP: " << ekf.covariance()
-       << "\n\n";
+    << "\n\n";
 
   // --- first predict/update ---
   ekf.predict(dynamics(imu1), dt, Q);
   cout << "--- After predict 1 ---\nX: " << ekf.state()
-       << "\nP: " << ekf.covariance() << "\n\n";
+    << "\nP: " << ekf.covariance() << "\n\n";
   ekf.update(h_gps, z1, R);
   cout << "--- After update 1 ---\nX: " << ekf.state()
-       << "\nP: " << ekf.covariance() << "\n\n";
+    << "\nP: " << ekf.covariance() << "\n\n";
 
   // --- second predict/update ---
   ekf.predict(dynamics(imu2), dt, Q);
   cout << "--- After predict 2 ---\nX: " << ekf.state()
-       << "\nP: " << ekf.covariance() << "\n\n";
+    << "\nP: " << ekf.covariance() << "\n\n";
   ekf.update(h_gps, z2, R);
   cout << "--- After update 2 ---\nX: " << ekf.state()
-       << "\nP: " << ekf.covariance() << "\n";
+    << "\nP: " << ekf.covariance() << "\n";
 
   return 0;
 }

--- a/gtsam/3rdparty/GeographicLib/CMakeLists.txt
+++ b/gtsam/3rdparty/GeographicLib/CMakeLists.txt
@@ -47,7 +47,7 @@ set (LIBVERSION_BUILD 17.1.2)
 string (TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 string (TOUPPER ${PROJECT_NAME} PROJECT_NAME_UPPER)
 
-cmake_minimum_required (VERSION 2.8.4) # This version was released 2011-02-16
+cmake_minimum_required (VERSION 3.5) # This version was released 2011-02-16
 
 # User-settable variables
 

